src/analysisd/decoders/winevtchannel.c

```diff
@@ -82,92 +99,40 @@ int DecodeWinevt(Eventinfo *lf){
     cJSON *json_eventdata_in = cJSON_CreateObject();
     cJSON *json_extra_in = cJSON_CreateObject();
+    cJSON *parsed_msg;
+    cJSON *audit_policy_management;
+    cJSON *event;
+    cJSON *msg_from_prov;
     int level_n;
     unsigned long long int keywords_n;
     XML_NODE node, child;
-    int num;
     char *extra = NULL;
     char *filtered_string = NULL;
     char *level = NULL;
     char *keywords = NULL;
-    char *msg_from_prov = NULL;
     char *returned_event = NULL;
-    char *event = NULL;
-    char *find_event = NULL;
-    char *end_event = NULL;
-    char *real_end = NULL;
-    char *find_msg = NULL;
-    char *end_msg = NULL;
-    char *next = NULL;
-    char *category = NULL;
+    char *severityValue = NULL;
     char *join_data = NULL;
     char *join_data2 = NULL;
-    char aux = 0;
+    char *get_field = NULL;
+    char *print_field = NULL;
     lf->decoder_info = winevt_decoder;
 
-    os_calloc(OS_MAXSTR, sizeof(char), event);
-    os_calloc(OS_MAXSTR, sizeof(char), msg_from_prov);
     os_calloc(OS_MAXSTR, sizeof(char), join_data);
 
-    find_event = strstr(lf->log, "Event");
-
-    if(find_event){
-        find_event = find_event + 8;
-        end_event = strchr(find_event, '"');
-
-        if(end_event){
-            real_end = end_event;
-            aux = *(end_event + 1);
-
-            if(aux != '}' && aux != ','){
-                while(1){
-                    next = real_end + 1;
-                    real_end = strchr(next,'"');
-
-                    if(real_end) {
-                        aux = *(real_end + 1);
-                        if (aux == '}' || aux == ','){
-                            end_event = real_end;
-                            break;
-                        }
-                    } else {
-                        mdebug1("Malformed 'Event' field");
-                        break;
-                    }
-                }
-            }
-
-            num = end_event - find_event;
+    parsed_msg = cJSON_Parse(lf->log);
 
-            if(num > OS_MAXSTR - 1){
-                mwarn("The event message has exceeded the maximum size.");
-                cJSON_Delete(json_system_in);
-                cJSON_Delete(json_event);
-                cJSON_Delete(json_eventdata_in);
-                cJSON_Delete(json_extra_in);
-                ret_val = 1;
-                goto cleanup;
-            }
-
-            memcpy(event, find_event, num);
-            event[num] = '\0';
-            find_event = NULL;
-            end_event = NULL;
-            real_end = NULL;
-            next = NULL;
-            aux = 0;
-        }
-    } else {
-        mdebug1("Malformed JSON output received. No 'Event' field found");
-    }
+    event = cJSON_GetObjectItem(parsed_msg, "Event");
 
     if(event){
-        if (OS_ReadXMLString(event, &xml) < 0){
+        print_field = cJSON_PrintUnformatted(event);
+
+        if (OS_ReadXMLString(print_field, &xml) < 0){
             first_time++;
```

Comment:
It has no sense to increase a variable eternally to check only if its value is higher than one and trigger a debug or a warning message. Why is not always a warning message? If it is the normal behavior, that means is not an error and should be a debug message for all cases.

Anyway, if you want to keep the behavior, please use a boolean condition.